firedrake/mg/ufl_utils.py

```diff
@@ -273,7 +298,7 @@ def coarsen_snescontext(context, self, coefficient_mapping=None):
     # Now that we have the coarse snescontext, push it to the coarsened DMs
     # Otherwise they won't have the right transfer manager when they are
     # coarsened in turn
-    for val in chain(coefficient_mapping.values(), (bc.function_arg for bc in problem.bcs)):
+    for val in chain(coefficient_mapping.values(), (bc._F.u if isinstance(bc, EquationBC) else bc.function_arg for bc in problem.bcs)):
```

Comment:
Similarly, if we update `coefficient_mapping` in `coarsen_bc()`, we shouldn't need to loop over `problem.bcs` here as `coefficient_mapping` will contain `bc.function_arg`.
I don't know why we need to add `bc._F.u` explicitly here. Isn't that in `coefficient_mapping`?